Add sample using path loss monitoring (#23534)
nrfconnect/sdk-nrf: samples/bluetooth/path_loss_monitoring/README.rst

After a successful connection, the Central measures path loss using the following formula:

.. code-block:: none

   path loss = Tx Power - RSSI

The application defines the following three signal quality zones based on the path loss value:

* Low path loss (strong signal) - **LED0** is lit.
* Medium path loss: - **LED1** is lit.
* High path loss (weak signal): - **LED2** is lit.

The Peripheral advertises its presence and accepts connections.

User interface
**************

Central
-------

* **LED0:** Indicates **Low** path loss (good signal).
* **LED1:** Indicates **Medium** path loss.
* **LED2:** Indicates **High** path loss (weak signal).

No buttons are used.

Peripheral
----------

No direct user interface.
The device advertises under the name ``path_loss_peripheral``.

Building and running
********************

.. |sample path| replace:: :file:`samples/bluetooth/path_loss_monitoring`
.. include:: /includes/build_and_run.txt

.. |sample_or_app| replace:: sample
.. |ipc_radio_dir| replace:: :file:`sysbuild/ipc_radio`
.. include:: /includes/ipc_radio_conf.txt

Testing
=======

|test_sample|

> 1. Program one DK with the `samples/bluetooth/path_loss_monitoring/central` program.
> #. Program another DK with the `samples/bluetooth/path_loss_monitoring/peripheral` program.

Reviewer: ```suggestion
1. Program one DK with the :file:`samples/bluetooth/path_loss_monitoring/central` program.
#. Program another DK with the :file:`samples/bluetooth/path_loss_monitoring/peripheral` program.
```

#. Power on both devices.
#. The Central will start scanning and connect automatically to the Peripheral.

> #. Once connected, observe the LEDs on the Central to evaluate signal strength:
>    * Bring the devices closer or move them farther apart to change the signal quality.
>    * Optionally, place an obstruction between the devices to see how this affects signal quality.

Reviewer: ```suggestion
#. Once connected, observe the LEDs on the Central to evaluate signal strength:

   * Bring the devices closer or move them farther apart to change the signal quality.
   * Optionally, place an obstruction between the devices to see how this affects signal quality.
```

#. Monitor the VCOM ports for logs showing connection status and real-time path loss.

Sample output
*************

The following output is logged from the Central device:

.. code-block:: console

   [00:00:01.000,000] <inf> central_unit: Starting path loss monitoring sample
   [00:00:01.500,000] <inf> central_unit: Scanning for peripherals...
   [00:00:03.020,000] <inf> central_unit: Found peripheral: FF:EE:DD:CC:BB:AA
   [00:00:03.050,000] <inf> central_unit: Connection to FF:EE:DD:CC:BB:AA
   [00:00:03.800,000] <inf> central_unit: Connected to FF:EE:DD:CC:BB:AA
   [00:00:03.810,000] <inf> central_unit: Path loss monitoring enabled
   [00:00:05.010,000] <inf> central_unit: Zone 0. Path loss: 30 dBm
   [00:00:06.010,000] <inf> central_unit: Zone 1. Path loss: 45 dBm
   [00:00:07.010,000] <inf> central_unit: Zone 2. Path loss: 63 dBm

Configuration
*************

|config|

Configuration options
*********************

Check and configure the following Kconfig options:

> * :kconfig:option:`CONFIG_BT_CENTRAL` - Enables the Bluetooth LE Central role (Central application).

Reviewer: Library options should be under 'Additional configuration'

###############################

.. contents::
   :local:
   :depth: 2

The path loss monitoring sample demonstrates how to evaluate Bluetooth® LE signal quality using the path loss monitoring feature.
It consists of a Central and a Peripheral device.
The Central continuously monitors the signal strength and classifies the connection quality using onboard LEDs.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

To run this sample, you must use two compatible development kits to run this sample: one programmed with the Central application, and the other with the Peripheral application.

Overview
********

This sample illustrates Bluetooth LE path loss monitoring by calculating the signal degradation between two connected devices.
The Central application scans for a Peripheral named ``path_loss_peripheral`` and connects to it.
After a successful connection, the Central measures path loss using the following formula:

.. code-block:: none

   path loss = Tx Power - RSSI

The application defines the following three signal quality zones based on the path loss value:

* Low path loss (strong signal) - **LED0** is lit.

> * Medium path loss: - **LED1** is lit.
> * High path loss (weak signal): - **LED2** is lit.

Reviewer: ```suggestion
* Medium path loss - **LED1** is lit.
* High path loss (weak signal) - **LED2** is lit.
```

.. _bluetooth_path_loss_monitoring:

> Bluetooth: Path Loss Monitoring

Reviewer 2: I'd suggest to have the name without all words starting with capital initial letter:
Path loss monitoring

###############################

.. contents::
   :local:
   :depth: 2

> The Path Loss Monitoring sample demonstrates how to evaluate Bluetooth® LE signal quality using the Path Loss Monitoring feature. It consists of a Central and a Peripheral device. The Central continuously monitors the signal strength and classifies the connection quality using onboard LEDs.

Reviewer 3: ```suggestion
The Path Loss Monitoring sample demonstrates how to evaluate Bluetooth® LE signal quality using the Path Loss Monitoring feature.
It consists of a Central and a Peripheral device. 
The Central continuously monitors the signal strength and classifies the connection quality using onboard LEDs.
```
New line for each sentence

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

You must use two compatible development kits to run this sample: one programmed with the Central application, and one with the Peripheral application.

Overview
********

> This sample illustrates Bluetooth® LE Path Loss Monitoring by calculating the signal degradation between two connected devices. The Central application scans for a Peripheral named ``Path_Loss_Peripheral`` and connects to it. After a successful connection, the Central measures path loss using the formula:

Reviewer 3: ```suggestion
This sample illustrates Bluetooth® LE Path Loss Monitoring by calculating the signal degradation between two connected devices.
The Central application scans for a Peripheral named ``path_loss_peripheral`` and connects to it. 
After a successful connection, the Central measures path loss using the formula:
```
New line for each sentence and use updated path_loss_peripheral name

::

   path loss = Tx Power - RSSI

The application defines three signal quality zones based on the path loss value:

- **Low path loss (strong signal):** LED0 is turned on.
- **Medium path loss:** LED1 is turned on.
- **High path loss (weak signal):** LED2 is turned on.

The Peripheral simply advertises its presence and accepts connections.

User interface
**************

Central:

> - **LED0:** Indicates **Low** path loss (good signal).
> - **LED1:** Indicates **Medium** path loss.
> - **LED2:** Indicates **High** path loss (weak signal).

Reviewer 2: ```suggestion
* **LED0:** Indicates **Low** path loss (good signal).
* **LED1:** Indicates **Medium** path loss.
* **LED2:** Indicates **High** path loss (weak signal).
```

No buttons are used.

Peripheral:

> - No direct user interface. The device advertises under the name ``Path_Loss_Peripheral``.

Reviewer 2: ```suggestion
No direct user interface.
The device advertises under the name ``Path_Loss_Peripheral``.
```

Building and running
********************

.. |sample path| replace:: :file:`samples/bluetooth/path_loss_monitoring`
.. include:: /includes/build_and_run.txt

.. |sample_or_app| replace:: sample
.. |ipc_radio_dir| replace:: :file:`sysbuild/ipc_radio`
.. include:: /includes/ipc_radio_conf.txt

Testing
=======

To test the sample:

> 1. Program one DK with the **Peripheral Path Loss Monitoring** firmware.
> 2. Program another DK with the **Central Path Loss Monitoring** firmware.
> 3. Power both devices.
> 4. The Central will start scanning and connect automatically to the Peripheral.
> 5. Once connected, observe the LEDs on the Central to evaluate signal strength:
>    - Bring the devices closer or move them farther apart to change the signal quality.
>    - Optionally place an obstruction between the devices to see how this affects signal quality.
> 6. Monitor the VCOM ports for logs showing connection status and real-time path loss.

Reviewer 2: ```suggestion
1. Program one DK with the **Peripheral Path Loss Monitoring** firmware.
#. Program another DK with the **Central Path Loss Monitoring** firmware.
#. Power on both devices.
#. The Central will start scanning and connect automatically to the Peripheral.
#. Once connected, observe the LEDs on the Central to evaluate signal strength:
   * Bring the devices closer or move them farther apart to change the signal quality.
   * Optionally place an obstruction between the devices to see how this affects signal quality.
6. Monitor the VCOM ports for logs showing connection status and real-time path loss.
```
By Peripheral and Central Path Loss Monitoring firmware, do you mean the samples? Which are?

Sample output
*************

Typical logs from the Central device:

.. code-block:: console

>    [00:00:01.000,000] <inf> Central_Unit: Starting Path Loss Monitoring sample
>    [00:00:01.500,000] <inf> Central_Unit: Scanning for peripherals...
>    [00:00:03.020,000] <inf> Central_Unit: Found peripheral: FF:EE:DD:CC:BB:AA
>    [00:00:03.050,000] <inf> Central_Unit: Connection to FF:EE:DD:CC:BB:AA
>    [00:00:03.800,000] <inf> Central_Unit: Connected to FF:EE:DD:CC:BB:AA
>    [00:00:03.810,000] <inf> Central_Unit: Path loss monitoring enabled
>    [00:00:05.010,000] <inf> Central_Unit: Zone 0. Path loss: 30 dBm
>    [00:00:06.010,000] <inf> Central_Unit: Zone 1. Path loss: 45 dBm
>    [00:00:07.010,000] <inf> Central_Unit: Zone 2. Path loss: 63 dBm

Reviewer 3: ```suggestion
   [00:00:01.000,000] <inf> central_unit: Starting Path Loss Monitoring sample
   [00:00:01.500,000] <inf> central_unit: Scanning for peripherals...
   [00:00:03.020,000] <inf> central_unit: Found peripheral: FF:EE:DD:CC:BB:AA
   [00:00:03.050,000] <inf> central_unit: Connection to FF:EE:DD:CC:BB:AA
   [00:00:03.800,000] <inf> central_unit: Connected to FF:EE:DD:CC:BB:AA
   [00:00:03.810,000] <inf> central_unit: Path loss monitoring enabled
   [00:00:05.010,000] <inf> central_unit: Zone 0. Path loss: 30 dBm
   [00:00:06.010,000] <inf> central_unit: Zone 1. Path loss: 45 dBm
   [00:00:07.010,000] <inf> central_unit: Zone 2. Path loss: 63 dBm
```

Configuration
*************

|config|

Configuration options
*********************

Check and configure the following Kconfig options:

> :kconfig:option:`CONFIG_BT_CENTRAL`
>     Enables the Bluetooth® LE Central role (Central application).

Reviewer 2: ```suggestion
* :kconfig:option:`CONFIG_BT_CENTRAL` - Enables the Bluetooth® LE Central role (Central application).
```
And so on, with no empty lines between the items.

:kconfig:option:`CONFIG_BT_PERIPHERAL`
    Enables the Bluetooth® LE Peripheral role (Peripheral application).

:kconfig:option:`CONFIG_BT_DEVICE_NAME`
    Sets the advertising name of the Peripheral. Default is ``Path_Loss_Peripheral``.

:kconfig:option:`CONFIG_BT_SCAN`
    Enables scanning for Bluetooth LE advertisements (Central only).

:kconfig:option:`CONFIG_BT_SCAN_FILTER_ENABLE`
    Enables name-based filtering for scan results (Central only).

:kconfig:option:`CONFIG_BT_PATH_LOSS_MONITORING`
    Enables the Bluetooth® LE Path Loss Monitoring feature on both devices.

:kconfig:option:`CONFIG_DK_LIBRARY`
    Enables the LEDs on the development kits.

Dependencies
************

This sample uses the following |NCS| libraries:

* :ref:`nrf_bt_scan_readme`
* :ref:`dk_buttons_and_leds_readme`

> It uses the following Zephyr libraries:

Reviewer 2: ```suggestion
It uses the following Zephyr library:
```

.. _bluetooth_path_loss_monitoring:

Bluetooth: Path loss monitoring
###############################

.. contents::
   :local:
   :depth: 2

The Path Loss Monitoring sample demonstrates how to evaluate Bluetooth® LE signal quality using the Path Loss Monitoring feature.

It consists of a Central and a Peripheral device.

> The Central continuously monitors the signal strength and classifies the connection quality using onboard LEDs.

Reviewer 2: All three could be in same paragraph, so remove the empty lines between these.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

You must use two compatible development kits to run this sample: one programmed with the Central application, and one with the Peripheral application.

Overview
********

This sample illustrates Bluetooth® LE Path Loss Monitoring by calculating the signal degradation between two connected devices.

The Central application scans for a Peripheral named ``path_loss_peripheral`` and connects to it.

> After a successful connection, the Central measures path loss using the formula:

Reviewer 2: These could also be just in one paragraph, so remove the empty lines between.

::

   path loss = Tx Power - RSSI

> The application defines three signal quality zones based on the path loss value:
> 
> - **Low path loss (strong signal):** LED0 is turned on.
> - **Medium path loss:** LED1 is turned on.
> - **High path loss (weak signal):** LED2 is turned on.

Reviewer 2: ```suggestion
The application defines the following three signal quality zones based on the path loss value:

* Low path loss (strong signal) - **LED0** is lit.
* Medium path loss - **LED1** is lit.
* High path loss (weak signal) - **LED2** is lit.
```

> The Peripheral simply advertises its presence and accepts connections.

Reviewer 2: ```suggestion
The Peripheral advertises its presence and accepts connections.
```

User interface
**************

> Central:

Reviewer 2: ```suggestion
Central
-------
```

* **LED0:** Indicates **Low** path loss (good signal).
* **LED1:** Indicates **Medium** path loss.
* **LED2:** Indicates **High** path loss (weak signal).

No buttons are used.

> Peripheral:

Reviewer 2: ```suggestion
Peripheral
----------
```

No direct user interface.

The device advertises under the name ``path_loss_peripheral``.

Building and running
********************

.. |sample path| replace:: :file:`samples/bluetooth/path_loss_monitoring`
.. include:: /includes/build_and_run.txt

.. |sample_or_app| replace:: sample
.. |ipc_radio_dir| replace:: :file:`sysbuild/ipc_radio`
.. include:: /includes/ipc_radio_conf.txt

Testing
=======

> To test the sample:

Reviewer 2: ```suggestion
To test the sample, complete the following steps:
```

1. Program one DK with the **Peripheral Path Loss Monitoring** sample.

> #. Program another DK with the **Central Path Loss Monitoring** sample.

Reviewer 2: Use links instead of the names in bold.
But wait, are there separate samples for Peripheral and Central?

   * Optionally place an obstruction between the devices to see how this affects signal quality.
#. Monitor the VCOM ports for logs showing connection status and real-time path loss.

Sample output
*************

Typical logs from the Central device:

.. code-block:: console

   [00:00:01.000,000] <inf> central_unit: Starting Path Loss Monitoring sample
   [00:00:01.500,000] <inf> central_unit: Scanning for peripherals...
   [00:00:03.020,000] <inf> central_unit: Found peripheral: FF:EE:DD:CC:BB:AA
   [00:00:03.050,000] <inf> central_unit: Connection to FF:EE:DD:CC:BB:AA
   [00:00:03.800,000] <inf> central_unit: Connected to FF:EE:DD:CC:BB:AA
   [00:00:03.810,000] <inf> central_unit: Path loss monitoring enabled
   [00:00:05.010,000] <inf> central_unit: Zone 0. Path loss: 30 dBm
   [00:00:06.010,000] <inf> central_unit: Zone 1. Path loss: 45 dBm
   [00:00:07.010,000] <inf> central_unit: Zone 2. Path loss: 63 dBm

Configuration
*************

|config|

Configuration options
*********************

Check and configure the following Kconfig options:

* :kconfig:option:`CONFIG_BT_CENTRAL` - Enables the Bluetooth® LE Central role (Central application).

> 

Reviewer 2: Remove the extra empty lines between the bullet items.

   * Bring the devices closer or move them farther apart to change the signal quality.
   * Optionally, place an obstruction between the devices to see how this affects signal quality.
#. Monitor the VCOM ports for logs showing connection status and real-time path loss.

Sample output
=============

The following output is logged from the Central device:

.. code-block:: console

   [00:00:01.000,000] <inf> central_unit: Starting path loss monitoring sample
   [00:00:01.500,000] <inf> central_unit: Scanning for peripherals...
   [00:00:03.020,000] <inf> central_unit: Found peripheral: FF:EE:DD:CC:BB:AA
   [00:00:03.050,000] <inf> central_unit: Connection to FF:EE:DD:CC:BB:AA
   [00:00:03.800,000] <inf> central_unit: Connected to FF:EE:DD:CC:BB:AA
   [00:00:03.810,000] <inf> central_unit: Path loss monitoring enabled
   [00:00:05.010,000] <inf> central_unit: Zone 0. Path loss: 30 dBm
   [00:00:06.010,000] <inf> central_unit: Zone 1. Path loss: 45 dBm
   [00:00:07.010,000] <inf> central_unit: Zone 2. Path loss: 63 dBm

Dependencies
************

This sample uses the following |NCS| libraries:

* :ref:`nrf_bt_scan_readme`
* :ref:`dk_buttons_and_leds_readme`

> It uses the following Zephyr library:

Reviewer 2: The list below is not a library, but a list of header files. Do you want to refer to the libraries?
